cal-itp/benefits
# Only show supported flows for In-person and Digital flows

closes #2396 

## What this PR does
- Adds filter() calls to both Eligibility form methods
- Updates test mock to return an array of flows
- Adds tests to the Digital and In-Person flows, testing a scenario in which an agency has 3 flows (digital, in-person and both( and the page should only show 2.

## How to test
- Log into Admin as a superuser and set different enrollment flows to be either in-person-only or digital-only.
- Log into Admin as a CS user to see that digital-only is filtered out
- Go to Benefits digital to see that in-person-only is filtered out

Note: Either digital-only or in-person-only HAVE to be checked. You can't set a flow to be neither.
